### PR TITLE
i#3044 AArch64 SVE codec: Add multi gpr memory load/stores

### PR DESCRIPTION
**Pull request by joshua-warburton**
This patch adds the appropriate macros, tests and codec entries to encode the following variants:
```
LD2B    { <Zt1>.B, <Zt2>.B }, <Pg>/Z, [<Xn|SP>, <Xm>]
LD3B    { <Zt1>.B, <Zt2>.B, <Zt3>.B }, <Pg>/Z, [<Xn|SP>, <Xm>]
LD4B    { <Zt1>.B, <Zt2>.B, <Zt3>.B, <Zt4>.B }, <Pg>/Z, [<Xn|SP>, <Xm>]
ST2B    { <Zt1>.B, <Zt2>.B }, <Pg>, [<Xn|SP>, <Xm>]
ST3B    { <Zt1>.B, <Zt2>.B, <Zt3>.B }, <Pg>, [<Xn|SP>, <Xm>]
ST4B    { <Zt1>.B, <Zt2>.B, <Zt3>.B, <Zt4>.B }, <Pg>, [<Xn|SP>, <Xm>]
```
issue: #3044

### CODE REVIEW

File: core/ir/opnd_api.h [hunk 1]
```diff
@@ -3268,6 +3268,15 @@ DR_API
 bool
 opnd_replace_reg(opnd_t *opnd, reg_id_t old_reg, reg_id_t new_reg);
 
+DR_API
+/**
+ * Creates a reg incremented  from an exiting \p opndby the \p increment value,
```
Comment (AssadHashmi) [inline core/ir/opnd_api.h:3273]: typo: existing
opndby all one word.

File: core/ir/opnd_shared.c [hunk 2]
```diff
@@ -1386,6 +1386,59 @@ opnd_replace_reg(opnd_t *opnd, reg_id_t old_reg, reg_id_t new_reg)
...
+opnd_t
+opnd_create_increment_reg(opnd_t opnd, uint increment)
+{
+    opnd_t inc_opnd DR_IF_DEBUG(= { 0 }); /* FIXME: Needed until i#417 is fixed. */
+    CLIENT_ASSERT(opnd_is_reg(opnd), "opnd_create_increment_reg: not a register");
+
+    reg_id_t reg = opnd.value.reg_and_element_size.reg;
+    reg_id_t min_reg = DR_REG_INVALID;
+    reg_id_t max_reg = DR_REG_INVALID;
+#ifdef AARCH64
```
Comment (AssadHashmi) [inline core/ir/opnd_shared.c:1398]: Looks like a significant chunk of this function, its main initialising code, is only compiled in for AArch64! Won't it fail for any architecture  other than AArch64? Currently none of the non-AArch64 regression tests call it, hence the full precommit pass.

As it's a fairly specific AArch64 SVE function it's probably best to `#ifdef` the whole function. Perhaps consider calling it `opnd_create_increment_sve_reg()` or something?

**Review by AssadHashmi — CHANGES_REQUESTED** [review]

**Review by AssadHashmi — APPROVED** [review]

**Comment by joshua-warburton** [comment]
Hey @derekbruening , merging seems to be broken. Do you know anything about the merge queue?

**Comment by derekbruening** [comment]
We enabled the merge queue feature yesterday to avoid simultaneous commits having to update and re-run tests.  Don't know anything about it beyond the doc https://github.blog/changelog/2023-02-08-pull-request-merge-queue-public-beta/

**Comment by derekbruening** [comment]
Is it this?
https://github.com/community/community/discussions/46757 
"make sure your Action workflow(s) are properly configured to trigger on merge_group event"

**Comment by joshua-warburton** [comment]
Seems like it might be the issue- I can see no evidence it is actually triggering the jobs. It also doesn't seem to give us the ability to modify the commit message on merge.

**Comment by derekbruening** [comment]
> It also doesn't seem to give us the ability to modify the commit message on merge.

Looks like we can now set it to use the PR title+desc -- maybe that is better in general as that can be reviewed prior to the squash:  https://github.blog/changelog/2022-08-23-new-options-for-controlling-the-default-commit-message-when-merging-a-pull-request/

I'm thinking it's worth trying these two things and if we still have problems we could then turn it off.  I will try them soon.

**Comment by joshua-warburton** [comment]
slight weirdness for us is that we have to escape the instruction names in our pull requests otherwise we get:
LD2B    { <Zt1>.B, <Zt2>.B }, <Pg>/Z, [<Xn|SP>, <Xm>]
and I wonder how that's going to end up in the commit message.

**Comment by derekbruening** [comment]
I disabled the merge queue for now -- does it let you merge this in the old way?

**Comment by derekbruening** [comment]
I've turned on the default final message as the pull request title and description.  Let's see how that goes.

**Pull request merged**